npm/pkg/dataplane/ipsets/ipsetmanager_linux.go

```diff
@@ -2,21 +2,59 @@ package ipsets
 
 import (
 	"fmt"
+	"regexp"
 
-	"github.com/Azure/azure-container-networking/log"
 	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/ioutil"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/parse"
 	"github.com/Azure/azure-container-networking/npm/util"
+	npmerrors "github.com/Azure/azure-container-networking/npm/util/errors"
+	"k8s.io/klog"
 )
 
 const (
-	maxTryCount                    = 1
-	deletionPrefix                 = "delete"
-	creationPrefix                 = "create"
+	azureNPMPrefix = "azure-npm-"
+
+	ipsetCommand        = "ipset"
+	ipsetSaveFlag       = "save"
+	ipsetRestoreFlag    = "restore"
+	ipsetCreateFlag     = "-N"
+	ipsetFlushFlag      = "-F"
+	ipsetAddFlag        = "-A"
+	ipsetDeleteFlag     = "-D"
+	ipsetDestroyFlag    = "-X"
+	ipsetExistFlag      = "--exist"
+	ipsetNetHashFlag    = "nethash"
+	ipsetSetListFlag    = "setlist"
+	ipsetIPPortHashFlag = "hash:ip,port"
+	ipsetMaxelemName    = "maxelem"
+	ipsetMaxelemNum     = "4294967295"
+
+	// constants for parsing ipset save
+	ipsetAddString           = "add"
+	ipsetCreateString        = "create"
+	ipsetSetListString       = "list:set"
+	ipsetNetHashString       = "hash:net"
+	ipsetIPPortHashString    = ipsetIPPortHashFlag
+	lengthForSuccessfulMatch = 2
+
+	// creator constants
+	maxTryCount                    = 3
```

Comment:
to my knowledge, there are only line-based errors in ipset restore. all lines before an error do get applied so we handle the error at a given line by restoring from the line after the failure (and removing any subsequent, related lines). So if an IP was malformed and we missed it, then the add for it will fail, we'll continue on, and the everything else will be applied as it should be

Reply:
I am not sure, but failure will be cascading (For example, if `create` failed, adding elements will be failed). 

The other concern is that I am not sure it is easy to resolve failed one in a next reconcile.
If you simply handle only failed one in a next reconcile, I think it will be fine.

Reply:
the current logic will handle cascading errors (e.g. if a create fails, then abandon all adds/deletes for that set)

eventually we will save these failures in a reconcile cache and handle separately